docs/topics/mta-web-tagging-an-application.adoc

```diff
@@ -58,4 +58,8 @@ A side drawer opens.
 +
 You can see the tags attached to the application.
 +
-The tags can be filtered by source and tag category. The sources are *Analysis* and *Manual*. The categories are displayed in a drop-down list, for example, *HTTP*, *MVC*, *Web*, *Observability*, *Persistence*, *Application Type*, *Data Center*.
\ No newline at end of file
+The tags can be filtered by source and tag category. The sources are *Analysis* and *Manual*. The categories are displayed in a drop-down list, for example, *HTTP*, *MVC*, *Web*, *Observability*, *Persistence*, *Application Type*, *Data Center*.
++
+If you have not applied the *Source* filter, all possible tag categories are enabled in the *Tag category* drop-down list.
++
+If you have applied the *Source* filter, only those tag categories that are relevant to the selected source are enabled in the *Tag category* drop-down list. For example, if the source is *Manual*, only the *Application Type* and *Data Center* tag categories are enabled.
```

Comment:
Hi @sbeskin-redhat ,
I see we haven't mentioned about the case where "No tags available" for example if we try to filter tags by `Manual source` and select not relevant tag category in this case User can `create a Tag` and attach to the application. We can add this as a note
![Screenshot from 2023-08-01 10-57-45](https://github.com/windup/windup-documentation/assets/87791967/88c6e598-cb5d-4d1e-933c-501c063422fe)

Reply:
@kpunwatk Thank you! 
Do I understand correctly that the option of creating a tag appears ONLY when no tags are available? Or is it always present?

Reply:
@sbeskin-redhat creating a tag option is present not only when no tags are available but also when filtering is applied and relevant tag categories attached are present.
Have attached the video to explain(Please find the attachment). Please do let me know if you need more info, will be happy to help !
[Screencast from 08-01-2023 03:48:16 PM.webm](https://github.com/windup/windup-documentation/assets/87791967/92a12422-15ce-4032-8363-6386a1072db7)

Reply:
@kpunwatk Thank you!
A couple of things are still not clear to me:
- In the video that you helpfully attached, the selected source is *Manual*, but all tag categories seem to be enabled. Shouldn't only *Application type* and *Data Center* be enabled?
- What happened before the 'No tags available' message appeared? You selected *Database* category?
- I didn't see the *Create tag* option until the 'No tags available' message appeared. When else is this option available?

Reply:
> @kpunwatk Thank you! A couple of things are still not clear to me:
> 
> * In the video that you helpfully attached, the selected source is _Manual_, but all tag categories seem to be enabled. Shouldn't only _Application type_ and _Data Center_ be enabled?

All tag categories _attached to an application_ are been reflected/displayed under Tag Categories(i.e. Both Manual and Source). But by enabled we mean user can filter tag categories those are attached.

> * What happened before the 'No tags available' message appeared? You selected _Database_ category?
Selected the Tag category which was for Analysis source and hence Database was not present for Manual source. In this case we provide an user additional feature to add manual tags in Database tag category.

> * I didn't see the _Create tag_ option until the 'No tags available' message appeared. When else is this option available?
Create tag option is present for following:
1. When user create an application without any manual tags and before running analysis. 
2. During filtering Manual Source user can add tags to the respective tag category present

Reply:
@kpunwatk 
So, is the following correct?
No matter which source the user selects, ALL tag categories that are attached to the application (both manually or through an analysis) are displayed in the *Tag category* list (and the user cannot distinguish between those that can be used for filtering and those that cannot). 
If the user selects the *Manual* source and then a tag category for which there are no tags from the manual source, he gets a *No tags available* message and the *Create* option. (And what if the user selects *Analysis* and a category where there are no tags, will the *Create* option appear?)

if the above is correct, then how does it resolve the MTA-549 issue? it says "Enable only tag categories relevant to the source in the 'Tag Category' drop down menu".

Reply:
Yes @sbeskin-redhat your following is correct
>And what if the user selects _Analysis_ and a category where there are no tags, will the _Create_ option appear?

Yes for this scenario too the  _Create tag_ option appear

> if the above is correct, then how does it resolve the MTA-549 issue? it says "Enable only tag categories relevant to the source in the 'Tag Category' drop down menu".

I see the issue regarding _Enable only tag categories relevant to the source in the 'Tag Category' drop down menu on application details page_ is still on `Assigned` state the and `fix version is 6.2.1`
https://issues.redhat.com/browse/MTA-415

Reply:
@kpunwatk 
Hi,
Has this feature ("Enable only tag categories relevant to the source in the 'Tag Category' drop down menu".) been implemented for 7.0? Should we finalize this PR?

Reply:
Hi @sbeskin-redhat,
Yes this feature is been implemented in MTA-7.0 only relevant tags re show. We can finalize this PR
![Screenshot from 2023-10-10 15-22-23](https://github.com/windup/windup-documentation/assets/87791967/717afe6e-f27f-4c8f-9bd6-eef114aadfac)

Reply:
@kpunwatk 
Thanks!
Do you know who could review this PR from the Devs team?

Reply:
@sbeskin-redhat I think Dylan, jeff, Ian, gilles can review this PR